lib/widgets/lightbox.dart

```diff
@@ -0,0 +1,200 @@
+                        .format(DateTime.fromMillisecondsSinceEpoch(widget.message.timestamp * 1000)),
+
+                      // Make smaller, like a subtitle
+                      style: themeData.textTheme.titleSmall!.copyWith(color: appBarForegroundColor)),
+                  ]
+                )),
+              centerTitle: false,
+              foregroundColor: appBarForegroundColor,
+              backgroundColor: appBarBackgroundColor,
+
+              // With Material 3, later, try removing these "transparent" overrides
+              shadowColor: Colors.transparent,
+              surfaceTintColor: Colors.transparent,
+            )
+          : null,
+        body: MediaQuery(
+          // Declare the vertical insets unconsumed by `appBar` and
+          // `bottomNavigationBar`, so we can position the image to not overlap
+          // any system intrusions, at least until the user zooms in. The top
+          // and bottom app bars overlay the pan-zoom layer in the Z direction;
+          // they don't affect anything in the pan-zoom layer's Y direction.
+          //
+          // Done by clobbering the ambient MediaQueryData (prepared by Scaffold
+          // for the `body`) with one taken from a BuildContext above the
+          // Scaffold.
+          data: MediaQuery.of(context),
+
+          child: GestureDetector(
+            behavior: HitTestBehavior.translucent,
+            onTap: _handleTap,
+            child: SizedBox.expand(
+              child: InteractiveViewer(
+                child: SafeArea(
+                  child: LightboxHero(
+                    message: widget.message,
+                    src: widget.src,
+                    child:
+                      Image.network(
+                        widget.src,
+                        filterQuality: FilterQuality.medium,
+                        headers: isUrlOnRealm(widget.src, store.account)
+                          ? Map.fromEntries([authHeader(store.account)])
+                          : null))))))),
+        bottomNavigationBar: _headerFooterVisible
+          ? BottomAppBar(
+              color: appBarBackgroundColor,
+
+              // With Material 3, later, try removing these "transparent" overrides
+              shadowColor: Colors.transparent,
+              surfaceTintColor: Colors.transparent,
+
+              child: Row(
+                children: [
+                  IconButton(
+                    tooltip: 'Copy link',
+                    icon: const Icon(Icons.copy),
+                    onPressed: () async {
+                      await Clipboard.setData(ClipboardData(text: widget.src));
+                      if (!context.mounted) return; // ignore: use_build_context_synchronously
+                      ScaffoldMessenger.of(context).showSnackBar(
+                        const SnackBar(
+                          behavior: SnackBarBehavior.floating,
+                          content: Text('Link copied')));
```

Comment:
> the system has its own "something was just copied to the clipboard" popup…

Ah, looks like the Android docs have guidelines for handling this:

https://developer.android.com/develop/ui/views/touch-and-input/copy-paste#duplicate-notifications
> In Android 12L (API level 32) and lower, we suggest alerting users that they’ve successfully copied by issuing a visual pop-up in-app widget (like Toasts or Snackbars) after copying.
>
> To avoid duplicate displays of information, we strongly recommend removing any pop-up widget shown after an in-app copy for Android 13 and higher.

How about we follow that? It seems easy.

Reply:
Sure, sounds good to me.